velox/connectors/hive/PartitionNameGenerator.h

```diff
@@ -61,17 +68,57 @@ class HivePartitionUtil {
   virtual std::string toPartitionString(Timestamp value, const TypePtr& type)
       const;
 
+  virtual std::string partitionName(
+      uint32_t partitionId,
+      const RowVectorPtr& partitionValues) const;
+
   /// Extract partition key-value pairs from a row vector at a specific row.
   ///
   /// @param partitionsVector A row vector containing partition columns.
   /// @param row The row index to extract partition values from.
   /// @return A vector of (column_name, formatted_value) pairs.
-  static std::vector<std::pair<std::string, std::string>>
+  virtual std::vector<std::pair<std::string, std::string>>
   extractPartitionKeyValues(
       const RowVectorPtr& partitionsVector,
-      vector_size_t row);
-};
+      vector_size_t row) const;
+
+  /// Returns the Hive string representation for null partition values.
```

Comment:
Can we replace getNullPartitionValue virtual method with a 'nullValue' string provided to the ctor?